src/node/util.ts

```diff
@@ -20,19 +22,23 @@ export const paths = getEnvPaths()
  * On MacOS this function gets the standard XDG directories instead of using the native macOS
  * ones. Most CLIs do this as in practice only GUI apps use the standard macOS directories.
  */
-function getEnvPaths(): Paths {
+export function getEnvPaths(): Paths {
   let paths: Paths
   if (process.platform === "win32") {
-    paths = envPaths("code-server", {
-      suffix: "",
-    })
+    paths = {
+      ...envPaths("code-server", {
+        suffix: "",
+      }),
+      runtime: tmpdir,
+    }
   } else {
     if (xdgBasedir.data === undefined || xdgBasedir.config === undefined) {
       throw new Error("No home folder?")
     }
     paths = {
       data: path.join(xdgBasedir.data, "code-server"),
       config: path.join(xdgBasedir.config, "code-server"),
+      runtime: xdgBasedir.runtime ? path.join(xdgBasedir.runtime, "code-server") : tmpdir,
```

Comment:
Originally I broke this out into and `if/else` statement but I feel like this is more condensed and doesn't sacrifice readability.

Reply:
Upon further thought, let's fall back to `temp` from the `env-paths` module instead of `tmpdir`. It uses a directory per-user so it'll ensure #2200 doesn't keep happening for users without a runtime dir specified.

Maybe something like this? Feel free to do it completely differently haha, I just started writing an example of what I meant and it turned into a whole thing.

```
const paths = envPaths("code-server", { suffix: "" })
const append = (p: string): string => path.join(p, "code-server")
switch (process.platform) {
  case "darwin":
    return {
      // envPaths uses native directories so force Darwin to use the XDG spec
      // to align with other CLI tools.
      data: xdgBasedir.data ? append(xdgBasedir.data) : paths.data,
      config: xdgBasedir.config ? append(xdgBasedir.config) : paths.config,
      // Fall back to temp if there is no runtime dir.
      runtime: xdgBasedir.runtime ? append(xdgBasedir.runtime) : paths.temp,
    }
  case "win32":
    return {
      data: paths.data,
      config: paths.config,
      // Windows doesn't have a runtime dir.
      runtime: paths.temp,
    }
  default:
    return {
      data: paths.data,
      config: paths.config,
      // Fall back to temp if there is no runtime dir.
      runtime: xdgBasedir.runtime ? append(xdgBasedir.runtime) : paths.temp,
    }
}
```